stdlib/domain.ml

```diff
@@ -139,7 +151,7 @@ let at_first_spawn f =
     raise (Invalid_argument "First domain already spawned")
   else begin
     let old_f = !first_spawn_function in
-    let new_f () = f (); old_f () in
+    let new_f () = old_f (); f () in
```

Comment:
Here it is important (and not obvious) that we are running domain-creation callbacks in FIFO order to be symmetric with the domain-exit callbacks in LIFO order. Could you write a comment that makes this intentional design choice explicit in the code?

Reply:
This makes sense, but I believe that the comment that you propose should be attached to `at_each_spawn` and `at_exit`. There's no dual for `at_first_spawn`. There is no strong reason for `at_first_spawn` to be FIFO apart from the fact that `at_each_spawn` is FIFO and choosing LIFO for `at_first_spawn` would be more confusing.